lua/apisix/plugins/wolf-rbac.lua

```
@@ -351,71 +359,154 @@ local function login()
+        return core.response.exit(500, fail_response("request to wolf-server failed!"))
     end
     if not body.ok then
-        core.log.error("user login [", request_debug, "] failed! response body:",
+        core.log.error("request [", request_debug, "] failed! response body:",
                        core.json.delay_encode(body))
-        return core.response.exit(200, {message = body.reason})
+        return core.response.exit(200, fail_response(body.reason))
     end
-    core.log.info("user login [", request_debug, "] success! response body:",
+
+    core.log.info("request [", request_debug, "] success! response body:",
                   core.json.delay_encode(body))
+    return body
+end
+
+local function wolf_rbac_login()
+    local args = get_args()
+    if not args then
+        return core.response.exit(400, fail_response("invalid request"))
+    end
+    if not args.appid then
+        return core.response.exit(400, fail_response("appid is missing"))
+    end
+
+    local appid = args.appid
+    local consumer = get_consumer(appid)
+    core.log.info("consumer: ", core.json.delay_encode(consumer))
+
+    local uri = consumer.auth_conf.server .. '/wolf/rbac/login.rest'
+    local headers = new_headers()
+    local body = request_to_wolf_server('POST', uri, headers, args)
 
     local userInfo = body.data.userInfo
     local wolf_token = body.data.token
 
     local rbac_token = create_rbac_token(appid, wolf_token)
-    core.response.exit(200, {rbac_token = rbac_token, user_info = userInfo})
+    core.response.exit(200, success_response(nil, {rbac_token = rbac_token, user_info = userInfo}))
+end
+
+local function get_wolf_token(ctx)
+    local url = ctx.var.uri
+    local action = ctx.var.request_method
+    local clientIP = core.request.get_ip(ctx)
+    local permItem = {action = action, url = url, clientIP = clientIP}
+    core.log.info("hit wolf-rbac change_password api")
+    local rbac_token = fetch_rbac_token(ctx)
```

we can return an error message

---

Is it not recommend to directly use `core.response.exit` to return error information in the called function? When I thought about it at that time, all the calls would no longer need to be judged. The code would look simpler